### PR TITLE
CI: add file format lint script to check manual line breaks and file encoding

### PR DESCRIPTION
**Pull request by CharLotteiu**
### What is changed, added or deleted? (Required)

Add file encoding check, manual line break check, and control character check.

The source code files are maintained in https://github.com/CharLotteiu/pingcap-docs-checks and referenced in the circleci config file using the `wget` command.

The script files of the following checks have been moved to https://github.com/CharLotteiu/pingcap-docs-checks:
- git conflicts
- unclosed tags
- manual line breaks
- file encoding
- control characters

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.0 (TiDB 5.0 versions)
- [x] v4.0 (TiDB 4.0 versions)
- [x] v3.1 (TiDB 3.1 versions)
- [x] v3.0 (TiDB 3.0 versions)
- [x] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [x] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by yikeke** [comment]
1. Suggest only outputting the `The above issues will cause website build failure. Please fix them.` hint once at the end: 
image

2. What does `sql-mode.md has been converted to UTF-8 without BOM` mean?

**Comment by CharLotteiu** [comment]
> 2. What does `sql-mode.md has been converted to UTF-8 without BOM` mean?

If you open `sql-mode.md` in the repo, you can see the display of the metadata is abnormal. This is because this file is encoded in the UTF-8 with BOM format. But most of our files are encoded in UTF-8 without BOM. The script helps to fix the display issue.

image

image

**Comment by yikeke** [comment]
Suggest providing more info to potential lint users: `sql-mode.md has been converted to UTF-8 without BOM` -> `sql-mode.md: this file's encoding has been converted to UTF-8 without BOM to avoid broken metadata display.`

File: scripts/file-format-lint.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
+import re, sys, os, codecs
```
Comment (yikeke) [inline scripts/file-format-lint.py:1]: Suggest maintaining this source code in your own repo and reference it in the circleci config file, so for further code changes, you don't have to open pull requests in 5 branches each time.

**Review by yikeke — APPROVED** [review]
LGTM

**Pull request merged**